Objects/typeobject.c

```diff
@@ -2325,11 +2325,14 @@ object_init(PyObject *self, PyObject *args, PyObject *kwds);
 static int
 type_init(PyObject *cls, PyObject *args, PyObject *kwds)
 {
+    PyTypeObject *type;
     int res;
 
     assert(args != NULL && PyTuple_Check(args));
     assert(kwds == NULL || PyDict_Check(kwds));
 
+    type = (PyTypeObject *)cls;
+
```

Comment:
Is this the proper way to get `type` from `cls`?  I tried `Py_TYPE(cls)` but that caused errors.

Reply:
Well, `Py_TYPE(cls)` is equivalent to `type(cls)` in Python (in this case that would give the metaclass), while what you're doing here is just reinterpreting the object as a type object. This form of cast is common in our C code.